js/autocomplete.js

```diff
@@ -226,99 +221,88 @@
       this.isOpen = false;
       this._mousedown = false;
     }
-    _renderDropdown(data, val) {
+    _renderDropdown(data, selectedValue) {
       this._resetAutocomplete();
-      let matchingData = [];
-      // Gather all matching data
-      for (let key in data) {
-        if (data.hasOwnProperty(key) && key.toLowerCase().indexOf(val) !== -1) {
-          let entry = {
-            data: data[key],
-            key: key
-          };
-          matchingData.push(entry);
 
-          this.count++;
-        }
-      }
+      let matchingData = data.filter(
+        (entry) => (entry.text || entry.id).toLowerCase().indexOf(selectedValue) !== -1
```

Comment:
wait, why `||`? Why there's "`text` or `id`"?

Reply:
Because the text property is optional and **id** is required but id can also be a string value and it searches first dor the text value and if none provided it searches the id value. The minimum entry could be {id: 'value1'}. I thought that it would then be better downward compatible, but we can change that if you want.

Reply:
Oh i see that there is **selectedValue** so id would be better yes. I will check it.